File: 프로그래머스/월간 코드 챌린지 시즌2/괄호 회전하기.py
# 괄호 회전하기
def solution(s):
    answer = 0
    def check(words):
        open_p,close_p = ['(','[','{'],[')',']','}']
        stack = []
        for w in words:
            if w in open_p:
                stack.append(w)
            else:
                if len(stack) == 0:
                    return False
                elif stack[-1] == open_p[close_p.index(w)]:
                    stack.pop()
                else:
                    return False
        if len(stack) == 0:
            return True
        else :
            return False
        
    for i in range(len(s)):
        words = s[i:]+s[:i]
        if check(words):
            answer += 1    
    return answer

File: 프로그래머스/월간 코드 챌린지 시즌2/괄호 회전하기.js
// 괄호 회전하기
function solution(s) {
    let answer = 0;
    const open = ['(','[','{'];
    const close = [')',']','}'];
    const check = function(par){
        let stack = [];
        par = par.split("");
        for(let i = 0; i < s.length; i++){
            if(open.includes(par[i]))
               stack.push(par[i]);
            else{
                if(stack.length > 0 && close.indexOf(par[i]) === open.indexOf(stack[stack.length-1]))
                    stack.pop();
                else return false;
            }
        }
        if(stack.length > 0) return false;
        return true;
    }
    for(let i = 0; i < s.length; i++){
        const st = s.slice(i) + s.slice(0,i);
        if(check(st)) answer += 1;
    }
    return answer;
}